lib/MusicBrainz/Server/Data/ArtistCredit.pm

```diff
@@ -25,26 +27,31 @@ sub get_by_ids
                 "JOIN artist ON artist.id=artist_credit_name.artist " .
                 "WHERE artist_credit IN (" . placeholders(@ids) . ") " .
                 "ORDER BY artist_credit, position";
+
     my %result;
     my %counts;
-    foreach my $id (@ids) {
-        my $obj = MusicBrainz::Server::Entity::ArtistCredit->new(id => $id);
-        $result{$id} = $obj;
```

Comment:
This was the problematic part: For every input ID, an output entity was created that remained empty if the database didn’t return anything about it. The `around` modifier in `Role::EntityCache` then cached these.